From b5e589fc385a2d4a62c6d91cefe6ad1b6bd9fa68 Mon Sep 17 00:00:00 2001
From: Chris Novakovic <chris@chrisn.me.uk>
Date: Sun, 8 Jul 2018 00:59:05 +0100
Subject: [PATCH] main: Add UseGatewaysAsTimeservers option

Add support for a new boolean option UseGatewaysAsTimeservers to the
[General] section of main.conf.

This is upstream commit 20f2041 backported for ConnMan 1.36.
---
 src/main.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/main.c b/src/main.c
index 318bf02..ad8595e 100644
--- a/src/main.c
+++ b/src/main.c
@@ -81,6 +81,7 @@ static struct {
 	char *vendor_class_id;
 	bool enable_online_check;
 	bool auto_connect_roaming_services;
+	bool use_gateways_as_timeservers;
 } connman_settings  = {
 	.bg_scan = true,
 	.pref_timeservers = NULL,
@@ -100,6 +101,7 @@ static struct {
 	.vendor_class_id = NULL,
 	.enable_online_check = true,
 	.auto_connect_roaming_services = false,
+	.use_gateways_as_timeservers = false,
 };
 
 #define CONF_BG_SCAN                    "BackgroundScanning"
@@ -120,6 +122,7 @@ static struct {
 #define CONF_VENDOR_CLASS_ID            "VendorClassID"
 #define CONF_ENABLE_ONLINE_CHECK        "EnableOnlineCheck"
 #define CONF_AUTO_CONNECT_ROAMING_SERVICES "AutoConnectRoamingServices"
+#define CONF_USE_GATEWAYS_AS_TIMESERVERS "UseGatewayAsTimeservers"
 
 static const char *supported_options[] = {
 	CONF_BG_SCAN,
@@ -140,6 +143,7 @@ static const char *supported_options[] = {
 	CONF_VENDOR_CLASS_ID,
 	CONF_ENABLE_ONLINE_CHECK,
 	CONF_AUTO_CONNECT_ROAMING_SERVICES,
+	CONF_USE_GATEWAYS_AS_TIMESERVERS,
 	NULL
 };
 
@@ -431,6 +435,13 @@ static void parse_config(GKeyFile *config)
 		connman_settings.auto_connect_roaming_services = boolean;
 
 	g_clear_error(&error);
+
+	boolean = __connman_config_get_bool(config, "General",
+				CONF_USE_GATEWAYS_AS_TIMESERVERS, &error);
+	if (!error)
+		connman_settings.use_gateways_as_timeservers = boolean;
+
+	g_clear_error(&error);
 }
 
 static int config_init(const char *file)
@@ -644,6 +655,9 @@ bool connman_setting_get_bool(const char *key)
 	if (g_str_equal(key, CONF_AUTO_CONNECT_ROAMING_SERVICES))
 		return connman_settings.auto_connect_roaming_services;
 
+	if (g_str_equal(key, CONF_USE_GATEWAYS_AS_TIMESERVERS))
+		return connman_settings.use_gateways_as_timeservers;
+
 	return false;
 }
 
-- 
2.7.4

